[Bug] Can't compile, RncLib warning/error

#### Describe the issue

Following the instructions, I attempted to compile the current git source, but I get an error related to RncLib, and it fails:

squigley@Latitude-E5530-non-vPro ~/CorsixTH $ mkdir build
squigley@Latitude-E5530-non-vPro ~/CorsixTH $ cd build
squigley@Latitude-E5530-non-vPro ~/CorsixTH/build $ cmake -DUSE_SOURCE_DATADIRS=ON .. 
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Note: SDL audio is enabled (default)
Note: FFmpeg video is enabled (default)
Note: FreeType2 is enabled (default)

Building common libraries
You have called ADD_LIBRARY for library RncLib without any source files. This typically indicates a problem with your CMakeLists.txt file
Building CorsixTH
CMake Error at CorsixTH/CMakeLists.txt:46 (add_executable):
  add_executable called with incorrect number of arguments


CMake Error at CorsixTH/CMakeLists.txt:57 (target_link_libraries):
  Cannot specify link libraries for target "CorsixTH" which is not built by
  this project.


-- Configuring incomplete, errors occurred!
See also "/home/squigley/CorsixTH/build/CMakeFiles/CMakeOutput.log".

I found the files under libs/rnc, and they don't configure correctly directly either:

squigley@Latitude-E5530-non-vPro ~/CorsixTH/libs/rnc $ cmake .
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
You have called ADD_LIBRARY for library RncLib without any source files. This typically indicates a problem with your CMakeLists.txt file
CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.10)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
-- Generating done
-- Build files have been written to: /home/squigley/CorsixTH/libs/rnc


#### Steps to Reproduce
1. git clone the source

2. mkdir build

3. cd build
4. cmake -DUSE_SOURCE_DATADIRS=ON ..


#### Expected Behaviour

cmake would complete, so I could compile the source.

#### System Information
**CorsixTH Version: git as of Dec 26th** file in .git includes " 541540bc9f95719269d9129196f53493ef3e988f Simon Quigley <squigley@Latitude-E5530-non-vPro.(none)> 1609043185 -0500	clone: from https://github.com/CorsixTH/CorsixTH.git"

**Operating System: Linux - 4.15.0-128-generic - x86_64** (Mint 19)

gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04) 

### Additional Info
*Paste any screen shots or other additional information that might help
illustrate the problem.*
[CMakeOutput.log](https://github.com/CorsixTH/CorsixTH/files/5748668/CMakeOutput.log)

--- a/.github/workflows/Linux.yml
+++ b/.github/workflows/Linux.yml
@@ -14,19 +14,32 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - lua: LuaJIT
+          - static_analysis: 0
+            docs: 0
+            cmake:
+            lua: LuaJIT
             luac: luac5.1
             packages: libluajit-5.1-dev luajit
             cmakejit: '-DWITH_LUAJIT=ON -DLUA_LIBRARY=/usr/lib/x86_64-linux-gnu/libluajit-5.1.so'
-          - lua: Lua 5.1
+          - static_analysis: 0
+            docs: 0
+            cmake:
+            lua: Lua 5.1
             luac: luac5.1
             packages: liblua5.1-dev lua5.1
-          - lua: Lua 5.3
+          - static_analysis: 1
+            docs: 1
+            cmake: 3.5
+            lua: Lua 5.3
             luac: luac5.3
             packages: liblua5.3-dev lua5.3
     name: Linux and Tests on ${{ matrix.lua }}
     steps:
       - uses: actions/checkout@v2  # Keeps the git OAuth token after checkout
+      - name: Setup CMake
+        uses: jwlawson/actions-setup-cmake@v1.5
+        with:
+          cmake-version: ${{ matrix.cmake }}
       - name: Install ${{ matrix.lua }}
         run: |
           sudo apt-get install ${{ matrix.packages }} luarocks
@@ -72,7 +85,7 @@ jobs:
           # Run lua unit tests
           busted --verbose --directory=CorsixTH/Luatest
       - name: Run simple code tests
-        if: matrix.lua == 'Lua 5.3'
+        if: matrix.static_analysis
         run: |
           # Check if there are trailing whitespaces.
           python3 scripts/check_trailing_whitespaces.py
@@ -99,7 +112,7 @@ jobs:
           yamllint --config-data "rules: {line-length: disable}" .github/workflows/Linux.yml
           shellcheck --shell sh scripts/macos_luarocks
       - name: Run clang code tests
-        if: matrix.lua == 'Lua 5.3'
+        if: matrix.static_analysis
         run: |
           # Check cpp format
           clang-format-9 -i CorsixTH/Src/*.cpp CorsixTH/Src/*.h AnimView/*.cpp \
@@ -109,11 +122,11 @@ jobs:
           clang-tidy-9 -p build --warnings-as-errors=\* \
             CorsixTH/Src/*.c CorsixTH/Src/*.cpp libs/rnc/*.cpp CorsixTH/SrcUnshared/main.cpp
       - name: Generate documentation
-        if: matrix.lua == 'Lua 5.3'
+        if: matrix.docs
         run: |
           cmake --build build/ --target -- doc
       - name: Upload documentation
-        if: github.ref == 'refs/heads/master' && github.repository == 'CorsixTH/CorsixTH' && matrix.lua == 'Lua 5.3'
+        if: github.ref == 'refs/heads/master' && github.repository == 'CorsixTH/CorsixTH' && matrix.docs
         run: |
           git config user.email "documentationbot@corsixth.com"
           git config user.name "Docs Bot"

--- a/CorsixTH/CMakeLists.txt
+++ b/CorsixTH/CMakeLists.txt
@@ -43,7 +43,7 @@ if(APPLE)
 elseif(MSVC)
   add_executable(CorsixTH CorsixTH.rc)
 else()
-  add_executable(CorsixTH)
+  add_executable(CorsixTH "")
 endif()
 
 # Ensure config.h is picked up by cmake - moving this into subdir cmake files will

--- a/libs/rnc/CMakeLists.txt
+++ b/libs/rnc/CMakeLists.txt
@@ -1,8 +1,6 @@
 project(RncLib)
 
-add_library(RncLib STATIC)
-
-target_sources(RncLib PRIVATE
+add_library(RncLib STATIC
   ${CMAKE_CURRENT_SOURCE_DIR}/rnc.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/rnc.h
 )

